myapp/app/controllers/application_controller.rb

```diff
@@ -1,2 +1,38 @@
 class ApplicationController < ActionController::Base
+
+  around_action :switch_locale
+
+  def switch_locale(&action)
+    locale = params[:locale] || I18n.default_locale
+    I18n.with_locale(locale, &action)
+  end
+
+  # keep locale in urls
+  def default_url_options
+    { locale: I18n.locale }
+  end
+
+
+  # unless Rails.env.development?
+    rescue_from Exception,                        with: :_render_500
+    rescue_from ActiveRecord::RecordNotFound,     with: :_render_404
+    rescue_from ActionController::RoutingError,   with: :_render_404
+  # end
```

Comment:
indents are not correct

```suggestion
  # unless Rails.env.development?
  rescue_from Exception,                        with: :_render_500
  rescue_from ActiveRecord::RecordNotFound,     with: :_render_404
  rescue_from ActionController::RoutingError,   with: :_render_404
  # end
```

Reply:
fixed